pkg/cmd/image/push.go

```diff
@@ -150,18 +149,7 @@ func Push(ctx context.Context, client *containerd.Client, rawRef string, options
 		return err
 	}
 
-	img, err := client.ImageService().Get(ctx, ref)
-	if err != nil {
-		return err
-	}
-	refSpec, err := reference.Parse(ref)
-	if err != nil {
-		return err
-	}
-	signRef := fmt.Sprintf("%s@%s", refSpec.String(), img.Target.Digest.String())
-	if err = signutil.Sign(signRef,
-		options.GOptions.Experimental,
-		options.SignOptions); err != nil {
+	if err = signutil.Sign(rawRef, options.GOptions.Experimental, options.SignOptions); err != nil {
```

Comment:
Probably no need to revert this, but use pushRef instead of ref

Reply:
That works for me.